prog/weaver/main.go

```diff
@@ -473,6 +470,20 @@ func canonicalName(f *mflag.Flag) string {
 	return ""
 }
 
+func checkpointFlags(router *weave.NetworkRouter) map[string]string {
+	flags := map[string]string{}
+	status := weave.NewNetworkRouterStatus(router)
+	for _, conn := range status.Connections {
+		if connectionName, ok := conn.Attrs["name"].(string); ok {
+			if _, encrypted := conn.Attrs["encrypted"]; encrypted {
+				connectionName = connectionName + " encrypted"
+			}
+			flags[connectionName] = "network"
```

Comment:
That is correct.  It seemed to me to be more common to have multiple distinct keys than multiple distinct values for a single key.